mautic/mautic
# Salesforce objects

Please answer the following questions:

| Q             | A
| ------------- | ---
| Bug fix?      | n
| New feature?  | y
| BC breaks?    | n
| Deprecations? | n
| Fixed issues  |  n
| User Documentation| https://github.com/mautic/documentation/pull/76

## Description
This PR, add functionality to Salesforce plugin.  The user will now be able to select what salesforce objects to pull data from and import them to Mautic contacts.

## Steps to test this PR
Apply this PR, in the salesforce configuration item you should now see the objects Lead and Contact.

The Salesforce CLI command will now fetch data from the selected objects and import them into Mautic Contacts. Pushing data will still continue to go into Salesforce Leads.

you need a secure site with domain name to test this: I have added a self signed certificate to my localhost and changed the name to a qualified domain name.

To test pulling leads run this command on command line:

php app/console mautic:integration:fetchleads --time-interval="29 days" --integration=Salesforce

This should import leads and/or contacts to mautic contacts

To test pushing lead activities
First create the salesforce object. you can follow steps here: https://github.com/mautic/documentation/pull/76/files

then run this command
php app/console mautic:integration:pushleadactivity --time-interval="29 days" --integration=Salesforce

this should push points, email activity, and form activity to the mautic object on salesforce to the leads that we have synced
make sure you update your namespace for the mautic object created in salesforce
